relay-server/src/endpoints/common.rs

```diff
@@ -337,17 +332,18 @@ pub async fn handle_envelope(
         )
     }
 
-    let buffer_guard = state.buffer_guard();
-    let mut managed_envelope = buffer_guard
-        .enter(
-            envelope,
-            state.outcome_aggregator().clone(),
-            state.test_store().clone(),
-            // It's not clear at this point which group this envelope belongs to.
-            // The decission will be made while queueing in `queue_envelope` function.
-            ProcessingGroup::Ungrouped,
-        )
-        .map_err(BadStoreRequest::QueueFailed)?;
+    if state.memory_checker().check_memory().is_exceeded() {
+        return Err(BadStoreRequest::QueueFailed);
```

Comment:
```suggestion
        // NOTE: Long-term, we should not reject the envelope here, but spool it to disk instead.
        // This will be fixed with the new spool implementation.
        return Err(BadStoreRequest::QueueFailed);
```